lib/gemstash/env.rb

```diff
@@ -115,12 +115,12 @@ def db
           db_path = base_file("gemstash.db")
 
           db = if RUBY_PLATFORM == "java"
-            Sequel.connect("jdbc:sqlite:#{db_path}", max_connections: 1)
+            Sequel.connect("jdbc:sqlite:#{db_path}", config[:db_connection_options].merge(max_connections: 1))
           else
-            Sequel.connect("sqlite://#{URI.escape(db_path)}", max_connections: 1)
+            Sequel.connect("sqlite://#{URI.escape(db_path)}", config[:db_connection_options].merge(max_connections: 1))
           end
         when "postgres", "mysql", "mysql2"
-          db = Sequel.connect(config[:db_url])
+          db = Sequel.connect(config[:db_url], config[:db_connection_options])
```

Comment:
What do you think about defaulting to the number of puma threads + 1 as @chriseckhardt ended up using? Perhaps `{ max_connections: config[:puma_threads] + 1 }.merge(config[:db_connection_options])`? This would mean adding a new `:puma_threads` config variable as well.